governance/program/src/state.rs

```diff
@@ -0,0 +1,247 @@
+    Proposal,
+
+    /// 3 - Proposal voting state account. Every Proposal account has exactly one ProposalState account
+    ProposalState,
+
+    /// 4 - Vote record account for a given Proposal.  Proposal can have 0..n voting records
+    VoteRecord,
+
+    /// 5 Custom Single Signer Transaction account which holds instructions to execute for Proposal
+    CustomSingleSignerTransaction,
+}
+
+impl Default for GovernanceAccountType {
+    fn default() -> Self {
+        GovernanceAccountType::Uninitialized
+    }
+}
+
+/// Vote  with number of votes
+#[derive(Clone, Debug, PartialEq)]
+pub enum Vote {
+    /// Yes vote
+    Yes(u64),
+
+    /// No vote
+    No(u64),
+}
+
+/// Governance Account
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct Governance {
+    /// Account type
+    pub account_type: GovernanceAccountType,
+
+    /// Voting threshold in % required to tip the vote
+    pub vote_threshold: u8,
+
+    /// Minimum slot time-distance from creation of proposal for an instruction to be placed
+    pub minimum_slot_waiting_period: u64,
+
+    /// Governance mint
+    pub governance_mint: Pubkey,
+
+    /// Council mint
+    pub council_mint: Option<Pubkey>,
+
+    /// Program ID that is governed by this Governance
+    pub program: Pubkey,
+
+    /// Time limit in slots for proposal to be open to voting
+    pub time_limit: u64,
+
+    /// Optional name
+    pub name: [u8; MAX_GOVERNANCE_NAME_LENGTH],
+
+    /// Running count of proposals
+    pub proposal_count: u32,
+}
+
+/// Governance Proposal
+#[derive(Clone)]
+pub struct Proposal {
+    /// Governance account type
+    pub account_type: GovernanceAccountType,
+
+    /// Governance account the Proposal belongs to
+    pub governance: Pubkey,
+
+    /// Proposal State account
+    pub state: Pubkey,
+
+    /// Mint that creates signatory tokens of this Proposal
+    /// If there are outstanding signatory tokens, then cannot leave draft state. Signatories must burn tokens (ie agree
+    /// to move instruction to voting state) and bring mint to net 0 tokens outstanding. Each signatory gets 1 (serves as flag)
+    pub signatory_mint: Pubkey,
+
+    /// Admin ownership mint. One token is minted, can be used to grant admin status to a new person.
+    pub admin_mint: Pubkey,
+
+    /// Mint that creates voting tokens of this Proposal
+    pub vote_mint: Pubkey,
+
+    /// Mint that creates evidence of voting YES via token creation
+    pub yes_vote_mint: Pubkey,
+
+    /// Mint that creates evidence of voting NO via token creation
+    pub no_vote_mint: Pubkey,
+
+    /// Used to validate signatory tokens in a round trip transfer
+    pub signatory_validation: Pubkey,
+
+    /// Used to validate admin tokens in a round trip transfer
+    pub admin_validation: Pubkey,
+
+    /// Used to validate voting tokens in a round trip transfer
+    pub vote_validation: Pubkey,
```

Comment:
We may also be able to get rid of these, since they increase the size of the struct by a lot.  Perhaps we just need to refactor some of the ownership / signature checking in `spl_token`

Reply:
Agree, I removed the validation accounts